core/src/main/java/com/google/ar/sceneform/ArSceneView.java

```diff
@@ -463,6 +465,10 @@ protected boolean onBeginFrame(long frameTimeNanos) {
         // Before doing anything update the Frame from ARCore.
         boolean arFrameUpdated = true;
         try {
+            // TODO: Understand why the camera texture name should be set not in the setSession method
```

Comment:
In the example in the ARCore repository the texture name is set on every frame too.

Reply:
Texture names should only be set once on a GL thread and it must done during onDrawFrame rather than onSurfaceCreated since the session is not guaranteed to have been initialized during the execution of onSurfaceCreated.